apps/api/src/app/integrations/usecases/remove-integration/remove-integration.usecase.ts

```diff
@@ -13,6 +13,7 @@ export class RemoveIntegration {
 
   async execute(command: RemoveIntegrationCommand) {
     try {
+      // TODO: We should check first if the Integration exists in the database
```

Comment:
Something I found.

Reply:
I am not entirely sure for 2 reasons.
1. due to the reason that the integration in the cache is stored as a query, even if we don't find integration by `command.integrationId` we still need to invalidate because it was not stored by `id` :(
2. regarding this one I am not sure, we could delete it without checking, and if it doesn't exist we will throw an exception from the DAL layer. that will basically mean 1 request to the DB and not 2.

Reply:
Regarding both points I am afraid that now with multiple integrations we would need to change that caching approach, as we will face situations of integrations for the same provider so the only way of differentiate them is by their id. 😕

Reply:
Will love to sync about it because i am not i am following the whole matter here